### PR TITLE
lr-vice - added additional targets to lr-vice

### PR DESCRIPTION
**Pull request by joolswills**
* added x64sc, x128, xpet, xplus4 and xvic on top of the default x64

### CODE REVIEW

**Comment by joolswills** [comment]
Did this the other day and forgot it wasn't quite complete. Need to fix up install still.

**Comment by joolswills** [comment]
Probably should have done more testing. Unfortunately the libretro makefiles are not smart enough to work without cleaning in-between, so this may be best done as separate modules that share code. Or just include maybe xvic and plus4. Not sure a libretro core for c128 or PET is that useful Vs the standalone emulator.

**Comment by joolswills** [comment]
Building xvic and xplus4 cores seems a reasonable compromise so I've gone with that for now vs having additional modules (which would have the overhead of downloading the source each time also).

**Comment by joolswills** [comment]
Ive changed my mind and I think as it requires a full build for each system, they are best implemented as separate modules, but which can share code with lr-vice. I'll revisit.

**Comment by joolswills** [comment]
I've looked into this again, and I don't like the duplication of module info etc with having lots of modules. Maybe I could implement some kind of sub-module function so a single scriptmodule can include additional modules whilst sharing the repository information etc.

I

**Comment by joolswills** [comment]
Added additional vice targets to the module.

File: scriptmodules/libretrocores/lr-vice.sh [hunk 1]
```diff
@@ -10,37 +10,57 @@
...
-    addSystem "c64"
+    local target
+    local name
+    for target in $(_get_targets_lr-vice); do
+        if [[ "$target" == "x64" ]]; then
+            name=""
+        else
+            name="-${target}"
+        fi
+        addEmulator 1 "$md_id${name}" "c64" "$md_inst/vice_${target}_libretro.so"
```
Comment (cmitu) [inline scriptmodules/libretrocores/lr-vice.sh:62]: Maybe move this into the `if` and set `x64` as default emulator
```` sh
if [[ "$target" == "x64" ]]; then
         addEmulator 1 "$md_id$" "c64" "$md_inst/vice_${target}_libretro.so"
 else
        addEmulator 0 "$md_id-${target}" "c64" "$md_inst/vice_${target}_libretro.so"
fi
````

File: scriptmodules/libretrocores/lr-vice.sh [hunk 2]
```diff
@@ -10,37 +10,57 @@
...
 }
 
 function build_lr-vice() {
-    make clean
-    make
-    md_ret_require="$md_build/vice_x64_libretro.so"
+    mkdir -p "$md_build/cores"
+    local target
+    for target in $(_get_targets_lr-vice); do
+        make clean
```
Comment (Gemba) [inline scriptmodules/libretrocores/lr-vice.sh:32]: Maybe it is advisable to put `make clean` outside the loop? Cuts compile time by roughly 2/3.

Comment (joolswills) [reply scriptmodules/libretrocores/lr-vice.sh:32]: I had that originally and it produced issues with the resulting cores. Maybe upstream libretro makefile has been improved, but feel free to test. Upstream mentions needing to clean before also afair.

Comment (joolswills) [reply scriptmodules/libretrocores/lr-vice.sh:32]: Actually just see my comments on the PR as I've already mentioned it.

Comment (joolswills) [reply scriptmodules/libretrocores/lr-vice.sh:32]: Here: https://github.com/RetroPie/RetroPie-Setup/pull/3126#issuecomment-623208782

But if anything has changed I can retest. Cheers.

Comment (Gemba) [reply scriptmodules/libretrocores/lr-vice.sh:32]: Compiled ok on my system but it seems sth is still lurking in the resulting builds: In their [README](https://github.com/libretro/vice-libretro/blame/master/README.md#L195) they note to run `make clean` before each building of an emulator.
So better keep this scriptmodule keep as-is.

**Pull request merged**